### PR TITLE
MRG: Fix for 1.8.0 dev

### PR DESCRIPTION
**Pull request by larsoner**
Better fix than #387. Passes on my machine™

### CODE REVIEW

File: sphinx_gallery/tests/test_gen_gallery.py [hunk 1]
```diff
@@ -68,9 +69,8 @@ def __init__(self, srcdir, confdir, outdir, doctreedir, buildername,
     def create_sphinx_app(self):
         # Avoid warnings about re-registration, see:
```
Comment (lesteve) [inline sphinx_gallery/tests/test_gen_gallery.py]: You probably want to remove the comment here and move it where it belongs.

**Comment by larsoner** [comment]
> Do you understand why it works?

Yes basically you need to use `docutils_namespace` context manager during each build. Before it was only being used when instantiating the `Sphinx` instance, which is not enough. And we weren't doing it for all `Sphinx` instances (one was missing for `tinybuild`).

Other comment addressed in latest commit.

**Comment by larsoner** [comment]
@lesteve okay to merge this? That way we stop the flow of daily Travis-is-sad emails :)

**Comment by larsoner** [comment]
> Before it was only being used when instantiating the Sphinx instance, which is not enough.

I thought this was the case but your commit appears to work on my machine with 1.8.0.dev. So this should be good to go with your reversion to the simpler version + tweaks

**Comment by larsoner** [comment]
That's great, I didn't like having to manually call the `__enter__` and `__exit__` methods

**Pull request merged**